lib/src/rules/enable_null_safety.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import '../analyzer.dart';
+
+const _desc = r'Do use sound null safety.';
+
+const _details = r'''
+
+**DO** use sound null safety, by not specifying a dart version lower than `2.12`.
+
+**BAD:**
+```dart
+// @dart=2.8
+a {
+}
+```
+
+**GOOD:**
+```dart
+b {
+}
+''';
+
+class EnableNullSafety extends LintRule implements NodeLintRule {
+  EnableNullSafety()
+      : super(
+            name: 'enable_null_safety',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this, context);
```

Comment:
I don't know how expensive the comment parsing below is but I wonder if it's worth bailing if non-nullability is enabled?

```dart
    if (context.isEnabled(Feature.non_nullable)) {
      return;
    }
```

Reply:
The comments have already been parsed and are in the AST so it's fairly inexpensive to iterate over them, but it wouldn't hurt to check.

Reply:
👍 

More specifically I mean the cost of the **_regexps_**.

Reply:
Thanks for hinting.

```dart
if (context.isEnabled(Feature.non_nullable)) {
  return;
}
```

doesn't work, giving all cases are ok (2 should fail).

On the other hand, there was also a suggestion to do something like:

```dart
if (line.replaceAll(' ', '').startsWith('//@dart=')) {
    // parse the version
}
```

if there is a concern about `regExp`.

Reply:
It might not be too bad, given that we only need to look at the comments before the first non-comment token in the file (modulo the corner case below). The performance is then dependent on the length of the file header and library documentation comments.